### PR TITLE
[MIGraphX EP] Set External Data Path

### PR DESCRIPTION
**Pull request by TedThemistokleous**
### Description

Changes to add in Set external data path for model weight files. Additional fixes to ensure this compiles off the latest v1.19 Onnxruntime

### Motivation and Context

Separate weights used for larger models (like stable diffusion) is motivation for this change set

### CODE REVIEW

**Comment by TedThemistokleous** [comment]
ping @PeixuanZuo @cloudhan @ytaous @snnn

File: onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc [hunk 1]
```diff
@@ -1297,6 +1299,7 @@ Status MIGraphXExecutionProvider::Compile(const std::vector<FusedNodeAndGraph>&
       if (!input_shape_match) {
         if (!load_precompiled_model(prog, load_compiled_model_, std::string{load_compiled_path_})) {
           LOGS_DEFAULT(VERBOSE) << "No Input shapes mismatch detected. Recompiling" << std::endl;
+          cmp_options.set_external_data_path(model_path_.parent_path().string());
```
Comment (tianleiwu) [inline onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc:1302]: Could you do some test model paths like "model.onnx" and "./model.onnx" ? Sometime, the path may not have parent path. If that is so, you might need something like `model_path_.has_parent_path() ? model_path_.parent_path() : std::filesystem::current_path()`

Comment (TedThemistokleous) [reply onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc]: Sure added

**Review by tianleiwu — APPROVED** [review]

**Comment by tianleiwu** [comment]
@TedThemistokleous,

I saw some build error in CI Pipleine:

https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1453986&view=logs&j=7536d2cd-87d4-54fe-4891-bfbbf2741d83&t=66420422-c7d6-5f71-625c-4b7851c9b9ba&l=2591

/onnxruntime_src/onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc:1302:23: error: ‘struct migraphx::api::onnx_options’ has no member named ‘set_external_data_path’

**Comment by TedThemistokleous** [comment]
You need to update to the latest MIGraphX that's been released.

**Pull request merged**